### ISSUE TITLE
Add support for cht-core v3.14

### ISSUE THREAD

**Issue by kennsippell** [medic/cht-conf-test-harness#150]
- [x] Some [standard stuff](https://github.com/medic/cht-conf-test-harness/blob/master/all-chts-bundle.js) for the new core version
- [x] Tests for new XPath functions

**Comment by kennsippell** [comment]
Hey @jkuester Let's begin the conversation of what it is going to take to support v3.14 wth the Enketo-core change.

**Comment by jkuester** [comment]
:rocket:  Your list here is great and covers pretty much all the points of concern that I have noted so far.  I will offer some thoughts below.  

Also relevant: https://github.com/medic/cht-conf-test-harness/issues/138

---

> Switch to pick up core's version of Enketo-core

You will know better then me, but I was particularly concerned about switching to use the version of enketo-core from whatever version of cht-core was selected. Offhand that sounds like it could involve considerable refactoring (since we might need an [enketo-adapter](https://github.com/medic/cht-conf-test-harness/blob/master/src/core-adapter.js) of some sort?).

---

> Awaiting the release of v3.14

 For the record, https://github.com/medic/cht-core/tree/6345-enketo-uplift contains the latest Enketo changes that will eventually end up in `3.14.0`.

---

> Can we detect parity differences between enketo on 3.13 and 3.14? Even stuff like "markdown support for questions" are likely to cause parity problems

This is a tough one.  Early on in the project, Craig suggested that we might look into some kind of "perceptional diffing" tool that could automatically detect differences between screenshots of 3.13 vs 3.14.  I still think this is a really good idea, but have not had time yet to do much investigation.  

---

> How do we find forms not using concat properly?

This is another tough one. Technically the [syntax for a concat statement](https://docs.getodk.org/form-operators-functions/?highlight=concat#concat) is pretty simple (each arg is separated with a comma). However, since you can nest `concat` functions and also include `if` statements as well, things get non-trivial really quickly.  It seems possible that we could come up with some regex that could do the parsing (similar to [what I did for required notes](https://gist.github.com/jkuester/292b0a584f7fc77a7d6c768f3c7c1ba3)), but I am not sure how high of confidence we could have in that....

**Comment by kennsippell** [comment]
@binokaryg @yrimal I've published `cht-conf-test-harness@2.4.0-beta.0` from branch `support-bikram` which addes 3.14 based on current cht-core master to support your work in https://github.com/medic/config-nssd/issues/25.  This doesn't include any of the expected complexity which we expect form 3.14's Enketo upgrade.

**Comment by yrimal** [comment]
Hi @kennsippell ,
When calling a form that uses bikram sambat, following error is thrown and we can't run test:
```
     Fill pregnancy_home_visit form for migrated.:
     Error: Evaluation failed: Error: Load Errors: ["FormLogicError: Could not evaluate: to-bikram-sambat( /model/instance[1]/pregnancy_home_visit/next_visit_date ), message: moment is not a function"]
    at FormWireup.render (webpack://cht-conf-test-harness/./src/form-host/wireup.js?:40:13)
    at window.loadXform (webpack://cht-conf-test-harness/./src/form-host/index.js?:12:33)
    at __puppeteer_evaluation_script__:1:85
```

You can see it in action in [25_Unit_Test](https://github.com/medic/config-nssd/tree/25_Unit_Test) if you run pregnancy home visit tests.

**Comment by kennsippell** [comment]
@yrimal I've published `2.4.0-beta.1`.  I've added a test for `to-bikram-sambat` and shouldn't crash.  The output in HTML looks a little funny still and I'm investigationg, but I'm hoping this unblocks.

**Comment by kennsippell** [comment]
Not going to work on the Enketo heavy lift since the Enketo release was bumped to cht-core 4.0

**Comment by kennsippell** [comment]
@yrimal Did you get a chance to test this? Is it working aspected to your knowledge?

**Comment by kennsippell** [comment]
Published as v2.4.1
